src/99-appendices/15-inheritance-principle.md

```diff
@@ -0,0 +1,260 @@
+# Appendix XV: The Inheritance Principle
+
+The [Inheritance Principle](#inheritance-principle) specifies how it is
+not strictly necessary to have data and metadata files with identical names
+apart from the [file extension](#definitions).
+It is possible to have a single metadata file with its contents applicable
+to multiple data files;
+it is additionally possible to have one data file where there are multiple
+metadata files from which metadata are applicable.
+This Appendix provides the precise definition of how this principle operates
+and the consequences of such,
+as reference for both software implementations and advanced BIDS users.
+
+## Rules
+
+1.  Any metadata file (such as `.json`, `.bvec` or `.tsv`) MAY be defined at any directory level.
+
+1.  For a given data file, any metadata file is applicable to that data file if:
+    1.  It is stored at the same directory level or higher;
+    1.  The metadata and the data filenames possess the same suffix;
+    1.  The metadata filename does not include any entity absent from the data filename.
+
+1.  A metadata file MUST NOT have a filename that would be otherwise applicable
+    to some data file based on rules 2.b and 2.c but is made inapplicable based on its
+    location in the directory structure as per rule 2.a.
+
+1.  If, for a given data file, multiple metadata files satisfy criteria 2.a-c above:
+
+    1.  The set of applicable metadata files is ordered as follows.
+        Within each level of the filesystem hierarchy independently,
+        applicable files are ordered from fewest to most entities;
+        each subsequent filename MUST possess a set of entities that is a
+        strict superset of the previous filename.
```

Comment:
<details>
<summary>Here is my initial concern/comment on ordering and ambiguity issue without reading/comprehending it in full.</summary> 

this does not address the ordering question for when number of entries is identical but different. e.g. `task-rest_bold.json` and `acq-tricky_bold.json`. And that is the main concern IMHO for introducing proposed relaxation.  Sure thing we could introduce some "alphabetic ordering" but then 
- not worth bothering with "ordered from fewest to most entities"
- "not all entities born alike" effect and preferential treatment solely based on alphabet order. IMHO very suboptimal

In either of those sorting orders, it might add substantial cognitive load on a human to properly "sort" files to be inherited.  And that is what I praise *original* IP for, that even given its "complexity" it is still human friendly with its limiting factor (no more than 1 to inherit at any given level) since it gives at least 80% coverage to existing use cases.
</details>

Although "cognitive load" aspect might still be considered high, I came to opinion that 
- this rule is Ok
- better than sorting I thought of
- it would provide some relaxation although still forbiding having complimentary e.g. `task-rest_bold.json` and `acq-tricky_bold.json` though. 

Taking it further from that point: For that latter case we could have removed this requirement ("subsequent ... is a strict superset ...") and require that for such cases of unclear ordering, order is arbitrary but metadata files MUST NOT have identical metadata fields overloaded (thus order would not matter).  

```suggestion
        if multiple applicable files have the same number of entities,
        they could be ordered arbitrarily among themselves and they MUST NOT
        contain overlapping metadata.
```

It would be up for validator to ensure adherence to such a principle. This would "relax" inheritance principle further and allow for arbitrary combinatorics if so desired for some studies, e.g. mix in of `_task-rest_bold.json` and `ses-pilot_bold.json`, `_space-X` and `_desc-Y` in https://github.com/bids-standard/bids-bep016/issues/32  etc.

Reply:
Personally, I would be in favour of this also. Partly for the capability as you describe, but also partly because it may actually facilitate *simplifying* the text. I'll make an attempt at explaining...

Currently, there is some clumsy text around attempting to describe the permissible ordering of sidecar files. It is necessary to describe such both with high accuracy in the appendix, and in the main text in a way that's comprehensible but not misleading.

With this proposed alteration, the text in the main specification could be simplified. It would basically be limited to rules 2a-c (written in a friendlier form); there would then be a "PS. There are circumstances where this isn't permitted due to ambiguity in the repeated definition of keys; see Appendix". So the ordering problem is not something that the typical reader would be expected to digest; whereas currently it's in there and it's not the easiest thing to follow.

For the IP itself, simplest would be to not permit *any* common keys within sidecar files within a single directory that are attributed to a common data file. So that would be changing your proposal to apply to all applicable metadata files within a single directory, not just those with the same number of entities. I *think* that in this hypothetical case, "number of entities" for ordering within a directory is an adequate substitution for rule 3a? I've failed to think of a limitation of such. Your more general solution of only forbidding collisions between sidecar files with the same number of entities is a little more powerful, it's just a little more complexity added to relevant softwares.

-----

Given that this isn't a trivial alteration of this PR, and it's far from guaranteed to achieve consensus given that there are already concerns with regard to complexity in this PR, I would suggest that this warrants its own PR. I would need to show the difference between the two not only in the text, but in the examples also: what's currently shown as impermissible would become permissible, and the complex example would need to be augmented to show two files in a directory with an equal number of entities being loaded in arbitrary order.

Reply:
> ...  to not permit any common keys within sidecar files within a single directory that are attributed to a common data file. 

Sounds good to me!

> I would suggest that this warrants its own PR.

well, although there is indeed incremental (relaxing restriction) in proposed change, its formulation would be in large conditioned on this PR, then we would have two PRs for people to digest, and first wait for this (already 6 months "in review") PR to be accepted.   But I am ok with that -- I do not have pressing need for it besides may be "be done with it once and for all"! ;)  So overall -- up to you.  As I have mentioned above, this formulation is "Ok" with me.

Reply:
thought to file a dedicate issue for that extra relaxation but then realized that it cannot then go as "wild" as you suggested:

> not permit any common keys within sidecar files within a single directory that are attributed to a common data file. 

be cause with this PR it would be permitted but overloaded through the order...  and would not address 
 https://github.com/bids-standard/bids-specification/issues/1182 . So I guess I will file it for the formulation I suggested -- only within the group of the same number of entities.

Reply:
filed a dedicated https://github.com/bids-standard/bids-specification/issues/1195 so let's resolve this conversation